src/Nancy/Helpers/ProxyNancyReferenceProber.cs

```diff
@@ -0,0 +1,18 @@
+#if !CORE
+namespace Nancy.Helpers
+{
+    using System;
+    using System.Reflection;
+    using Nancy.Extensions;
+
+    internal class ProxyNancyReferenceProber : MarshalByRefObject

```

Comment:
@thecodejunkie But this is a class defined at the namespace level, you can't have a private class unless it's inside another class. Would you rather have this class put inside the `AppDomainAssemblyCatalog` class?